### PR TITLE
firefox: Add option to build with GTK3

### PR DESCRIPTION
**Pull request by kirelagin**
Does anyone care about GTK3 as much as I do?
Because, you know, pixel-perfect scrolling is awesome, I enjoy it every single time I use zathura and I miss it so much in my web browser!

So, this PR makes Firefox build with GTK3 in a pretty straightforward way. To be honest, I have no idea what is the status of GTK3 support in Firefox right now, there is a [huge tracker](https://bugzilla.mozilla.org/show_bug.cgi?id=627699) but it is hard to tell what is expected to work. I definitely have a problem with themes, Oxygen is not being applied (that’s probably #5053) and there is another strange thing: all the popup menus are black.

Any ideas?

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -10836,6 +10836,7 @@ let
   firefox = callPackage ../applications/networking/browsers/firefox {
     inherit (gnome) libIDL;
     inherit (pythonPackages) pysqlite;
+    enableGTK3 = false;
```
Comment (edolstra) [inline pkgs/top-level/all-packages.nix]: This line is unnecessary.

Comment (kirelagin) [inline pkgs/top-level/all-packages.nix]: @edolstra I believe GTK3 should be enabled by default once everything is polished, so this line is a reminder.

**Comment by vcunat** [comment]
About status: I can't find much about it, but I think the state shouldn't be too bad, as almost a year ago the gtk3 port had [optimistic target of Firefox 32](https://lists.fedoraproject.org/pipermail/desktop/2014-June/009913.html).

**Comment by kirelagin** [comment]
I’ve been using this Firefox since I opened the PR, and didn’t notice any issues other than the one mentioned above, black pop-up menus. I still have no idea what to do with that.

**Comment by vcunat** [comment]
Maybe try with ``gtk3_16``, as #5053 suggests some related changes around 3.14.

**Comment by kirelagin** [comment]
In #5053 it is mentioned only that 3.14 makes Adwaita the default theme, I doubt this will help ;). I’ll try.

**Comment by kirelagin** [comment]
Yep, the black menus are gone. I suspect this has something to do with colours, as now all the menus are plain white with black text (not sure what colour or texture they are expected to have).

**Comment by vcunat** [comment]
It was supposed to be the default *since* 3.14, so 3.16 should be the same in that respect.

**Pull request closed without merging**